control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -955,33 +955,63 @@ func (r *reconciler) reconcileIngressController(ctx context.Context, hcp *hyperv
 	return errors.NewAggregate(errs)
 }
 
-func (r *reconciler) reconcileOAuthClientSecrets(ctx context.Context, hcp *hyperv1.HostedControlPlane) error {
+func (r *reconciler) reconcileAuthOIDC(ctx context.Context, hcp *hyperv1.HostedControlPlane) error {
 	var errs []error
 	if !util.HCPOAuthEnabled(hcp) &&
-		len(hcp.Spec.Configuration.Authentication.OIDCProviders) != 0 &&
-		len(hcp.Spec.Configuration.Authentication.OIDCProviders[0].OIDCClients) > 0 {
-		for _, oidcClient := range hcp.Spec.Configuration.Authentication.OIDCProviders[0].OIDCClients {
-			var src corev1.Secret
-			err := r.cpClient.Get(ctx, client.ObjectKey{Namespace: hcp.Namespace, Name: oidcClient.ClientSecret.Name}, &src)
+		len(hcp.Spec.Configuration.Authentication.OIDCProviders) != 0 {
+
+		// Copy issuer CA configmap into openshift-config namespace
+		provider := hcp.Spec.Configuration.Authentication.OIDCProviders[0]
+		if provider.Issuer.CertificateAuthority.Name != "" {
+			name := provider.Issuer.CertificateAuthority.Name
+			var src corev1.ConfigMap
+			err := r.cpClient.Get(ctx, client.ObjectKey{Namespace: hcp.Namespace, Name: name}, &src)
 			if err != nil {
-				errs = append(errs, fmt.Errorf("failed to get OIDCClient secret %s: %w", oidcClient.ClientSecret.Name, err))
-				continue
+				errs = append(errs, fmt.Errorf("failed to get issuer CA configmap %s: %w", name, err))
 			}
-			dest := corev1.Secret{
+			dest := corev1.ConfigMap{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:      oidcClient.ClientSecret.Name,
+					Name:      name,
 					Namespace: "openshift-config",
 				},
 			}
 			_, err = r.CreateOrUpdate(ctx, r.client, &dest, func() error {
 				if dest.Data == nil {
-					dest.Data = map[string][]byte{}
+					dest.Data = map[string]string{}
 				}
-				dest.Data["clientSecret"] = src.Data["clientSecret"]
+				dest.Data["ca-bundle.crt"] = src.Data["ca-bundle.crt"]
```

Comment:
if there was an error retrieving the src configMap at 968, wouldn't src be `nil`?

Reply:
I think 972-988 should be inside an else branch when `err == nil` since we are not returning.